dvc/project.py

```diff
@@ -267,6 +273,7 @@ def run(self,
 
         self._check_output_duplication(stage.outs)
         self._check_circular_dependency(deps, outs)
+        self._check_argument_duplication(outs)
```

Comment:
Looks nice! Thank you! I hope you don't mind a few more comments :slightly_smiling_face:  :

1) Let's move this up before Stage.loads(), so that this issue is detected early.
2) `outs` might not contain normalized paths (i.e. './file' and 'file'), so you need to normalize them before putting into set().
3) `outs_no_cache` should be checked along with `outs`.
4) could you please add a similar check for deps, since it is prone to such duplication as well?